src/Stryker.Core/Stryker.Core/Initialisation/ProjectMutator.cs

```diff
@@ -38,10 +39,25 @@ public IMutationTestProcess MutateProject(StrykerOptions options, IReporter repo
             // initial test
             input.InitialTestRun = initialisationProcess.InitialTest(options);
 
+            // Enrich test projects info with unit tests
+            EnrichTestProjectsWithTestInfo(input.InitialTestRun, input.TestProjectsInfo);
+
             // mutate
             process.Mutate();
 
             return process;
         }
+
+        private void EnrichTestProjectsWithTestInfo(InitialTestRun initialTestRun, TestProjectsInfo testProjectsInfo)
+        {
+            foreach (var unitTest in initialTestRun.Result.VsTestDescriptions.Select(desc => desc.Case))
+            {
+                var testFile = testProjectsInfo.TestFiles.SingleOrDefault(testFile => testFile.FilePath == unitTest.CodeFilePath);
+                if (testFile is not null)
```

Comment:
Maybe log a debug warning if we can't find the corresponding testfile? As that should not happen